packages/docs/celo-codebase/protocol/identity/accounts.md

```diff
@@ -0,0 +1,75 @@
+
+### Key Rotation for Consensus
+
+Key rotation for Consensus is a bit trickier. Let's assume that a validator is currently elected, actively signing consensus messages and wants to rotate their validator signer key. What they should do is the following:
+
+- Bring up a new validator node and create a new Validator signer key (VSK) on it.
+- Authorize the new Validator signer key with the Account key to overwrite the old Validator signer key.
+
+```bash
+celocli account:authorize --from <ACCOUNT_KEY_ADDRESS> --role validator --signer <NEW_VSK_ADDRESS> --signature <PROOF_OF_NEW_VSK_POSSESSION>
+```
+
+**This should also update your ECDSA public key automatically.**
+
+- Update your BLS key used for consensus. Rotating both the ECDSA and BLS keys is recommended.
+
+```bash
+celocli validator:update-bls-public-key --from <NEW_VSK_ADDRESS> --blsKey <NEW_BLS_KEY> --blsPop <PROOF_OF_NEW_BLS_KEY_POSSESSION>
+```
+
+- Your old node will continue to sign consensus messages for the current epoch, but upon the next epoch will find itself no longer authorized.
+- Your new node will find itself unauthorized, but will be authorized to sign as the next epoch begins.
+
+**Waiting for the epoch to flip before bringing down the retired validator node is essential to zero-downtime rotation when elected.**
+
+Please see the [Running a Validator](https://docs.celo.org/getting-started/baklava-testnet/running-a-validator) section for details on ECDSA and BLS key generation, authorization, and proof of possession.
+
+# Key Security
+
+It is evident that the Account key is the most sensitive key. As it should be used quite infrequently, it is highly recommended for important Account keys (i.e. Validators or accounts with high balances) to remain as secure as possible. At the minimum, we recommend them to be offline, ideally in cold storage or on a hardware wallet.
+
+# Key Usage in Protocol
+
+### Celo Registry Contracts
+
+`Accounts` Entrypoint for users to create an account and authorize _unique_ addresses which correspond to signing keys
+
+- requires user is _not_ a validator if authorizing Validator key and _not_ updating ECDSA public key
+- requires user is a validator if authorizing Validator key and updating ECDSA public key
+
+`LockedGold` Expects user to be registered account signing with Account key
+
+`Governance` Expects user to be registered account signing with Vote key (or account key)
+
+`Election` Expects user to be registered account signing with Vote key (or account key)
+
+`Validators` Expects user to be registered account signing with Validator key (or account key)
+
+`Attestations` Expects issuers to be registered accounts; expects completers to be signing with Attestation key (or account key)
+
+### Celo Blockchain Client
+
+`Consensus` Expects participants to be registered accounts and signing with Validator key
+
+`Precompiles` Keeps track of current validator set and proof of possession using authorized Validator key addresses
```

Comment:
This is a useful list a reference to developers, but I wouldn't include it here because this location is slightly higher level. Also, this list will surely become out-of-date unless it is kept somewhere it will be read frequently.